Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/Allocator/Jobs/Overlay/SendOverlay.hs

```diff
@@ -38,72 +41,113 @@ sendOverlayToDriver ::
-        else do
-          unless (null driverIds) $ mapM_ (sendPaymentPendingOverlay (fromIntegral limit) overlayKey udf1) driverIds
-          case rescheduleInterval of
-            Just interval -> do
-              ReSchedule . addUTCTime (fromIntegral interval) <$> getCurrentTime
-            Nothing -> return Complete
-    -- driversToNofify <- B.runInReplica $ findAllNotNotifiedWithDues merchantId transporterConfig.minOverlayGap 50
-    -- driverFees <- findPendingPaymentByDrivers (driversToNofify <&> (.driverId))
-    -- if null driverFees
-    --     then return Complete
-    --     else do
-    --         groupBy (\a b -> )
-    DOverlay.InactiveAutopay -> do
-      return Complete
+      jobId = id
+
+  driverIds <- getBatchedDriverIds merchantId jobId jobData.condition jobData.freeTrialDays jobData.timeDiffFromUtc jobData.driverPaymentCycleDuration jobData.driverPaymentCycleStartTime
+  logInfo $ show driverIds
+  driverIdsLength <- getSendOverlaySchedulerDriverIdsLength merchantId jobId
+
+  if driverIdsLength > 0
+    then do
+      mapM_ (sendOverlayAccordingToCondition jobData.condition jobData.overlayKey jobData.udf1) driverIds
+      ReSchedule . addUTCTime 180 <$> getCurrentTime -- 3 minutes
+    else do
+      unless (null driverIds) $ mapM_ (sendOverlayAccordingToCondition jobData.condition jobData.overlayKey jobData.udf1) driverIds
+      case jobData.rescheduleInterval of
+        Just interval -> do
+          lastScheduledTime <- getLastScheduledJobTime merchantId jobId jobData.scheduledTime jobData.timeDiffFromUtc
+          let newScheduledTime = addUTCTime (fromIntegral interval) lastScheduledTime
+          setLastScheduledJobTime merchantId jobId newScheduledTime
+          pure (ReSchedule newScheduledTime)
+        Nothing -> return Complete
   where
-    sendPaymentPendingOverlay limit overlayKey udf1 driverId = do
+    sendOverlayAccordingToCondition condition overlayKey udf1 driverId = do
       driver <- QP.findById driverId >>= fromMaybeM (PersonDoesNotExist driverId.getId)
-      pendingDriverFees <- QDF.findAllPendingAndDueDriverFeeByDriverId driverId
-      let manualDues = sum $ map (\dueInvoice -> SLDriverFee.roundToHalf (fromIntegral dueInvoice.govtCharges + dueInvoice.platformFee.fee + dueInvoice.platformFee.cgst + dueInvoice.platformFee.sgst)) $ filter (\due -> due.status == DDF.PAYMENT_OVERDUE) pendingDriverFees
-      when (manualDues > limit) $ sendOverlay driver overlayKey udf1 0
+      case condition of
+        DOverlay.PaymentOverdueGreaterThan limit -> do
+          manualDues <- getManualDues driverId
+          when (manualDues > fromIntegral limit) $ sendOverlay driver overlayKey udf1 0
+        DOverlay.PaymentOverdueBetween rLimit lLimit -> do
+          manualDues <- getManualDues driverId
+          when ((fromIntegral rLimit <= manualDues) && (manualDues <= fromIntegral lLimit)) $ sendOverlay driver overlayKey udf1 0
```

Comment:
`But where is the check of greater than limit or in b/w l and r?`

- Here are checking the condition